test/hotspot/jtreg/gc/metaspace/TestMetaspacePerfCounters.java

```diff
@@ -183,10 +183,45 @@
  * @run main/othervm -XX:+UsePerfData -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC gc.metaspace.TestMetaspacePerfCounters
  */
 
+class PerfCounterSnapshot {
+    private static long getMinCapacity(String ns) throws Exception {
+        return PerfCounters.findByName(ns + ".minCapacity").longValue();
+    }
+
+    private static long getCapacity(String ns) throws Exception {
+        return PerfCounters.findByName(ns + ".capacity").longValue();
+    }
+
+    private static long getMaxCapacity(String ns) throws Exception {
+        return PerfCounters.findByName(ns + ".maxCapacity").longValue();
+    }
+
+    private static long getUsed(String ns) throws Exception {
+        return PerfCounters.findByName(ns + ".used").longValue();
+    }
+
+    public long minCapacity;
+    public long maxCapacity;
+    public long capacity;
+    public long used;
+
+    public void get(String ns) throws Exception {
+        minCapacity = getMinCapacity(ns);
+        maxCapacity = getMaxCapacity(ns);
+        used = getUsed(ns);
+        capacity = getCapacity(ns);
+    }
+
+    public boolean equals(Object object) {
+        PerfCounterSnapshot other = (PerfCounterSnapshot)object;
```

Comment:
I think this should not be `equals` since it's not a proper implementation of `Object.equals`.  Come up with a different name (`equivalent`?  `consistent`?) that takes another snapshot object.